GitUI/UserControls/AvatarControl.cs

```diff
@@ -64,7 +62,7 @@ public void ClearCache()
             this.InvokeAndForget(async () =>
                     {
                         AvatarService.UpdateAvatarProvider();
-                        await _avatarCacheCleaner.ClearCacheAsync().ConfigureAwait(true);
+                        _avatarCacheCleaner.ClearCache();
                         await UpdateAvatarAsync().ConfigureAwait(false);
                     });
```

Comment:
This should be run in the background. `UpdateAvatarAsync` switches to the UI thread itself.
```
            ThreadHelper.FileAndForget(async () =>
                    {
                        AvatarService.UpdateAvatarProvider();
                        _avatarCacheCleaner.ClearCache();
                        await UpdateAvatarAsync();
                    });
```